src/protocols/secure_channel/RendezvousParameters.h

```diff
@@ -90,10 +97,12 @@ class RendezvousParameters
     RendezvousParameters & SetRemoteNodeId(NodeId nodeId)
     {
         mRemoteNodeId.SetValue(nodeId);
+        mHasCSRNonce = true;
```

Comment:
This is odd: why does csrnonce depend on remote node id?

if this is the case, you may as well use `mRemoteNodeId.HasValue()` for your bool logic. However I claim this is wrong.

Reply:
sorry this is wrong, thanks for catching this. Probably a bad merge. the bool mHasCSRNonce will be set to true only if the CSRNonce is set. Rectified it.